app/models/concerns/taskable.rb

```diff
@@ -18,7 +18,7 @@ module Taskable
     scope :for_circle, ->(circle) { joins(:working_group).where(working_groups: { circle_id: circle.id } ) }
     scope :for_project, ->(project) { where(project_id: project.id) }
 
-    scope :with_role, ->(role) { where(task_roles: {role_type: klass::Role.role_types[role]}) }
+    scope :with_role, ->(role) { where("#{klass.to_s.downcase}_roles" => {role_type: klass::Role.role_types[role]}) }
```

Comment:
Try using `klass::Role.table_name` instead of `#{klass.to_s.downcase}_roles` here. While your solutio will work most of the time, I can see it break when someone puts a  `table_name :some_other_table` in of the Role models. With my suggestion, Rails also does the string munging work for you and is forwards compatible.